src/JestProcessManagement/JestProcess.ts

```diff
@@ -85,16 +83,15 @@ export class JestProcess {
 
   public stop(): Promise<void> {
     return new Promise(resolve => {
-      this.stopRequested = true
       this.keepAliveCounter = 1
-      this.resolve = resolve
+      this.stopResolveCallback = resolve
       this.jestSupportEvents.clear()
       this.runner.closeProcess()
 
       // As a safety fallback to prevent the stop from hanging, resolve after a timeout
       // TODO: If `closeProcess` can be guarenteed to always resolve, remove this
       setTimeout(() => {
-        if (this.resolve === resolve) {
+        if (this.stopResolveCallback === resolve) {
           resolve()
         }
```

Comment:
just notice this, what happens when they are not the same? is there any reason we should not resolve() regardless? I would also imagine the way to trigger this race condition is to issue 2nd `stop()` before the first one completed: the 2nd stop() will override the `this.stopResolveCallback` (how about calling it "processStopped"?), we should have a test and handle this scenario, otherwise, I suspect the first stop() will probably hang, even with the timeout...

Reply:
Ah yes, that would cause a hang. It should be safe to remove the conditional and simplify it to just `resolve()` to avoid the hang, since multiple `resolve` calls are no-ops.

Ideally (perhaps in a future change), it would be good to refactor it away from creating the promise in `stop`, and create and store the promise inside the `startRunner` call instead. That would eliminate that ugly `resolve` ivar. Probably pair it with the `runner` ivar, as something like an `onExit` promise ivar. That could also mean wrapping/replacing/deprecating the `onExitCallback` API at the same time to keep things consistent.

Reply:
ok, i removed condition and added test case